File: problems/1414_find_the_minimum_number_of_fibonacci_numbers_whose_sum_is_k/Solution.java
import java.util.*;

class Solution {
    public int findMinFibonacciNumbers(int k) {
        List<Integer> fibs = new ArrayList<>(Arrays.asList(1, 1));
        while (fibs.get(fibs.size() - 1) < k) {
            int sz = fibs.size();
            fibs.add(fibs.get(sz - 1) + fibs.get(sz - 2));
        }
        int count = 0;
        for (int i = fibs.size() - 1; i >= 0 && k > 0; i--) {
            if (fibs.get(i) <= k) {
                k -= fibs.get(i);
                count++;
            }
        }
        return count;
    }
}

File: problems/1414_find_the_minimum_number_of_fibonacci_numbers_whose_sum_is_k/solution.py
class Solution:
    def findMinFibonacciNumbers(self, k: int) -> int:
        fibs = [1, 1]
        while fibs[-1] < k:
            fibs.append(fibs[-1] + fibs[-2])
        count = 0
        for i in range(len(fibs) - 1, -1, -1):
            if fibs[i] <= k:
                k -= fibs[i]
                count += 1
                if k == 0:
                    break
        return count
